test/benchmark/api_v1_1_tests.py

```diff
@@ -93,50 +133,77 @@ def test_prepare_discovery(self):
     def test_discovery(self):
         """ Wait for discovery finished """
         self.case_recorder.write_event('start all discovery')
-        self.__task = WorkerThread(AMQPWorker(queue=QUEUE_GRAPH_FINISH, \
-                                              callbacks=[self.handle_discovery_finish]), \
-                                   'benchmark discovery')
-        def start(worker, id):
-            worker.start()
-        tasks = WorkerTasks(tasks=[self.__task], func=start)
-        tasks.run()
-        tasks.wait_for_completion(timeout_sec=1200)
-        assert_false(self.__task.timeout, \
-            message='timeout waiting for task {0}'.format(self.__task.id))
-
-    def handle_discovery_finish(self, body, message):
-        routeId = message.delivery_info.get('routing_key').split('graph.finished.')[1]
-        Workflows().workflows_get()
-        workflows = loads(self.client.last_response.data)
-
-        for w in workflows:
-            definition = w['definition']
-            injectableName = definition.get('injectableName')
-            if injectableName == "Graph.SKU.Discovery":
-                graphId = w['context'].get('graphId')
-                if graphId == routeId:
-                    status = body.get('status')
-                    if status == 'succeeded':
-                        message.ack()
-                        self.__discovered += 1
-                        self.case_recorder.write_event('finish discovery {0}'
-                                .format(self.__discovered))
-                        break
+        self._wait_until_graph_finish('Graph.SKU.Discovery', 1200)
 
-        if self.node_count == self.__discovered:
-            self.__task.worker.stop()
-            self.__task.running = False
-            self.__discovered = 0
-            self._collect_case_data()
-            LOG.info('Fetch discovery log finished')
+    @test(groups=["test-bm-discovery-post"],
+            depends_on_groups=["test_discovery_add_obm"])
+    def test_discovery_post(self):
+        pass
 
 @test(groups=["benchmark.bootstrap"])
 class BenchmarkBootstrapTests(BenchmarkTests):
     def __init__(self):
-        BenchmarkTests.__init__(self,'bootstrap')
-        self.__task_worker = None
-        self.__discovered = 0
+        BenchmarkTests.__init__(self, 'bootstrap')
+        self.__base = os.getenv('RACKHD_BASE_REPO_URL', \
+             'http://{0}:{1}'.format(HOST_IP, HOST_PORT))
+        self.__os_repo = os.getenv('RACKHD_CENTOS_REPO_PATH', \
+            self.__base + '/repo/centos/7')
+
+    def __post_workflows(self, graph_name, body):
```

Comment:
Could this be moved into the workflows_tests.py module so other workflow related tests can pick up the retry/status checks?